### PR TITLE
R20.1 runtime estimate

### LINKED ISSUE #240: Add runtime estimate to validation and optimization
Thor already gives back a runtime estimate. We should add the estimate to "Validating..." and "Optimizing..." so users know what to expect

### PR DESCRIPTION
**Pull request by jake-kirkland-teton**
This is functioning as expected. Ended up taking the simpler route of tacking it on the end of the label since the QML was being uncooperative.

Closes #240

### CODE REVIEW

File: SmartSlicePlugin/SmartSliceCloudConnector.py [hunk 1]
```diff
@@ -228,6 +229,13 @@ def __call__(self, job: pywim.http.thor.JobInfo) -> bool:
         elif job.status == pywim.http.thor.JobInfo.Status.running and self.connector.status not in (SmartSliceCloudStatus.BusyOptimizing, SmartSliceCloudStatus.BusyValidating):
             self.connector.status = self._previous_status
             self.connector.updateSliceWidget()
+
+        if job.status == pywim.http.thor.JobInfo.Status.running and self.connector.status is SmartSliceCloudStatus.BusyOptimizing:
+            Logger.log("d", Duration(job.runtime_estimate).getDisplayString(DurationFormat.Format.Seconds))
```
Comment (dougkenik) [inline SmartSlicePlugin/SmartSliceCloudConnector.py:234]: You still want this here?

File: SmartSlicePlugin/SmartSliceCloudConnector.py [hunk 2]
```diff
@@ -228,6 +229,13 @@ def __call__(self, job: pywim.http.thor.JobInfo) -> bool:
...
+            remaining_time = Duration(job.runtime_estimate)
+            if int(remaining_time.getDisplayString(DurationFormat.Format.Seconds)) > 60:
```
Comment (bradymadams) [inline SmartSlicePlugin/SmartSliceCloudConnector.py:236]: `job.runtime_estimate` is already an integer I believe, so I don't think it should be necessary to go int -> str -> int.

File: SmartSlicePlugin/SmartSliceCloudConnector.py [hunk 3]
```diff
@@ -228,6 +229,13 @@ def __call__(self, job: pywim.http.thor.JobInfo) -> bool:
         elif job.status == pywim.http.thor.JobInfo.Status.running and self.connector.status not in (SmartSliceCloudStatus.BusyOptimizing, SmartSliceCloudStatus.BusyValidating):
             self.connector.status = self._previous_status
             self.connector.updateSliceWidget()
+
+        if job.status == pywim.http.thor.JobInfo.Status.running and self.connector.status is SmartSliceCloudStatus.BusyOptimizing:
+            Logger.log("d", Duration(job.runtime_estimate).getDisplayString(DurationFormat.Format.Seconds))
+            remaining_time = Duration(job.runtime_estimate)
```
Comment (bradymadams) [inline SmartSlicePlugin/SmartSliceCloudConnector.py:235]: `runtime_estimate` is the overall runtime, not the remaining time. We do return the estimated remaining time, but it looks like the pywim JobInfo class needs to be updated to include it.

Do we want the overall runtime estimate or remaining runtime estimate?

Comment (dougkenik) [reply SmartSlicePlugin/SmartSliceCloudConnector.py:235]: I was just thinking of this. We want remaining time

Comment (bradymadams) [reply SmartSlicePlugin/SmartSliceCloudConnector.py:235]: I went ahead and updated pywim to include it. `JobInfo.runtime_remaining`.

File: SmartSlicePlugin/SmartSliceCloudConnector.py (hunk 1 above)
Comment (jake-kirkland-teton) [reply SmartSlicePlugin/SmartSliceCloudConnector.py:234]: No. Good catch.

File: SmartSlicePlugin/SmartSliceCloudConnector.py (hunk 2 above)
Comment (jake-kirkland-teton) [reply SmartSlicePlugin/SmartSliceCloudConnector.py:236]: Fair enough. Though this is actually a good point, the check is no longer needed, it was in there for when we were considering displaying validate time. It has been removed.

File: SmartSlicePlugin/SmartSliceCloudConnector.py (hunk 3 above)
Comment (jake-kirkland-teton) [reply SmartSlicePlugin/SmartSliceCloudConnector.py:235]: Perfect. I have that swapped over to remaining time.

**Pull request merged**